src/controllers/expenses.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const expensesService = require('../services/expenses.js');
+const usersService = require('../services/users.js');
+
+const getAll = (req, res) => {
+  const queryParams = req.query;
+
+  if (queryParams) {
+    const filteredExpenses = expensesService.getFilteredExpenses(queryParams);
+
+    res.send(filteredExpenses);
+
+    return;
+  }
+
+  const expenses = expensesService.getExpenses();
+
+  res.send(expenses);
+};
+
+const getOne = (req, res) => {
+  const { expenseId } = req.params;
+  const foundExpense = expensesService.getExpense(expenseId);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.statusCode = 200;
+  res.send(foundExpense);
+};
+
+const add = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  if (!userId || !spentAt || !title || !amount || !category || !note) {
```

Comment:
I did it like this:
`  if (!user || Object.keys(req.body) < 6) {`
   ` res.sendStatus(400);`

    `return;`
  `}`
  
  Using this method, you can avoid destructuring and reduce code